packages/extension/src/components/molecules/NFTCard/NFTCard.tsx

```diff
@@ -0,0 +1,127 @@
+import { NFTImage } from '../../atoms';
+import { TruncatedText } from '../../atoms';
+import { NFTDetails } from '../../organisms';
+
+export interface NFTCardProps {
+  NFT: AccountNFToken;
+}
+
+const Transition = forwardRef(function Transition(
+  props: TransitionProps & {
+    children: ReactElement;
+  },
+  ref: Ref<unknown>
+) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+export const NFTCard: FC<NFTCardProps> = ({ NFT }) => {
+  const { getNFTData } = useLedger();
+  const [NFTData, setNFTData] = useState<NFTData | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const handleViewNFTClick = useCallback(() => {
+    setDialogOpen(true);
+  }, []);
+
+  const handleCloseDialog = useCallback(() => {
+    setDialogOpen(false);
+  }, []);
+
+  useEffect(() => {
+    const fetchNFTImg = async () => {
+      try {
+        setIsLoading(true);
+        const nftData = await getNFTData({ NFT });
+        setNFTData(nftData);
+      } catch (error) {
+        setNFTData({
+          NFTokenID: NFT.NFTokenID,
+          description: NFT.URI ? convertHexToString(NFT.URI) : 'No data'
+        });
+        Sentry.captureException(error);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchNFTImg();
+  }, [getNFTData, NFT]);
+
+  const handleTokenIdClick = (tokenId: string) => {
+    navigator.clipboard.writeText(tokenId);
+  };
+
+  if (!NFTData) return null;
+
+  return (
+    <>
+      <Dialog
+        open={dialogOpen}
+        onClose={handleCloseDialog}
+        fullScreen
+        TransitionComponent={Transition}
+      >
+        <NFTDetails NFTData={NFTData} handleClose={handleCloseDialog} />
+      </Dialog>
+      <Paper
+        elevation={5}
+        style={{
+          padding: '10px',
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+          marginBottom: '10px'
+        }}
+      >
+        <ListItem
+          style={{
+            flexDirection: 'column',
+            textAlign: 'center'
+          }}
+        >
+          {isLoading ? (
+            <CircularProgress data-testid="progressbar" />
+          ) : (
+            <NFTImage imageURL={NFTData.image} height={150} width={150} />
+          )}
+          <Tooltip title={NFTData.NFTokenID}>
+            <TruncatedText
+              onClick={() => handleTokenIdClick(NFTData.NFTokenID)}
+              text={NFTData.NFTokenID}
+              sx={{ fontSize: '14px', color: 'grey', marginTop: '10px', cursor: 'pointer' }}
+            />
+          </Tooltip>
+          <TruncatedText
+            text={NFTData.name}
+            sx={{ fontSize: '16px', color: 'white', marginTop: '10px' }}
+            data-testid="nft_name"
+          />
+          <TruncatedText
+            text={NFTData.description}
+            sx={{ fontSize: '14px', color: 'grey', marginTop: '10px' }}
+            maxLength={200}
```

Comment:
I see what you want to achieve here.
I think it can be improved by letting the CSS magic to handle the display of the multi lines text. Setting the string max length manually is "clunky" in my view.
I would make an update to your current `TruncatedText` component and also create a new component to handle multi lines text truncation.

Here is a code sandbox example with the updated `TruncatedText` and `TruncatedMultiLinesText` component:
https://codesandbox.io/p/sandbox/boring-heisenberg-qrrv9w

The `TruncatedMultiLinesText` component accepts ` maxLines` prop.

In both component you can initialize the `text` prop with `""` to avoid checking the `null` value later. It's cleaner this way.

Let me know what you think.

Reply:
Thanks a lot, I have refactored the `TruncatedText` component and introduced the `TruncatedMultiLinesText` 👍 It seems great

Reply:
Thanks a lot! I have refactored the `TruncatedText` component and introduced the `TruncatedMultiLinesText` 👍.  

I think it's great